localstack/utils/aws/arns.py

```diff
@@ -88,24 +89,19 @@ def _resource_arn(name: str, pattern: str, account_id: str, region_name: str) ->
 #
 
 
-def iam_role_arn(role_name: str, account_id: str) -> str:
+def iam_role_arn(role_name: str, account_id: str, region_name: str) -> str:
     if not role_name:
         return role_name
-    if role_name.startswith("arn:aws:iam::"):
+    if re.match(f"{ARN_PARTITION_REGEX}:iam::", role_name):
         return role_name
-    return "arn:aws:iam::%s:role/%s" % (account_id, role_name)
-
-
-def iam_policy_arn(policy_name: str, account_id: str) -> str:
-    if ":policy/" in policy_name:
-        return policy_name
-    return "arn:aws:iam::{}:policy/{}".format(account_id, policy_name)
+    return "arn:%s:iam::%s:role/%s" % (get_partition(region_name), account_id, role_name)
 
 
 def iam_resource_arn(resource: str, account_id: str, role: str = None) -> str:
     if not role:
         role = f"role-{resource}"
-    return iam_role_arn(role_name=role, account_id=account_id)
+    # Only used in tests, so we can hardcode the region for now
+    return iam_role_arn(role_name=role, account_id=account_id, region_name="us-east-1")
```

Comment:
That might become an issue when we do multi-partition tests! I can fix it as part of the test work though.